src/ec2/model.ts

```diff
@@ -63,6 +64,7 @@ export class Ec2ConnectionManager {
         const hasProperPolicies = await this.hasProperPolicies(selection.instanceId)
 
         if (!isInstanceRunning) {
+            telemetry.record({ result: 'Failed', reason: 'EC2SSMStatus' })
```

Comment:
You shouldn't need to do this if the error is bubbled up.

Reply:
same for the other `record({ result: 'Failed' })` cases here, right?

Reply:
Removed them, and retested. Found a missing `await` while I was at it.